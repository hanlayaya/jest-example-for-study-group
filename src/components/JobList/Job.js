import React from 'react';
import PropTypes from 'prop-types';

const Job = ({ title, company }) => {
  return (
    <div>
      <h1>{title}</h1>
      <h2>{company}</h2>
    </div>
  );
}

Job.propTypes = {
  title: PropTypes.string.isRequired,
  company: PropTypes.string,
};

Job.defaultProps = {
  company: '(unknown)',
};

Job.displayName = 'Job';

export default Job;
