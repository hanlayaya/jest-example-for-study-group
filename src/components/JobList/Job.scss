.job {
  text-align: left;
  border-bottom: 1px solid #eee;
  margin-bottom: 10px;
  padding-bottom: 10px;
}

.skill-tags {
  margin-top: 5px;
}

.skill-tag {
  display: inline-block;
  margin-right: 5px;
  padding: 2px;
  border: 1px solid #efefef;
  border-radius: 4px;
  background-color: #eee;
  color: #333;
  font-size: 11px;
}
