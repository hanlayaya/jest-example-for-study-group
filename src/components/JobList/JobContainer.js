import { connect } from 'react-redux';
import idx from 'idx';
import Job from './Job';

const mapStateToProps = (state, ownProps) => {
  return {
    job: idx(state, _ => _.youratorJobs.byId[ownProps.id].items) || {},
  };
};

export default connect(
  mapStateToProps,
)(Job);
