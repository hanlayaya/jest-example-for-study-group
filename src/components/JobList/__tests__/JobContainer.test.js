import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import storeFake from '../../../utils/testUtils';
import Job from '../Job';
import JobContainer from '../JobContainer';

describe('container <Job />', () => {
  let subject;
  let Component;
  let JobComponent;

  beforeEach(() => {
    const store = storeFake({
      youratorJobs: {
        list: {
          isFetching: false,
          didInvalidate: false,
          items: [1, 2],
        },
        byId: {
          1: {
            items: {
              id: 1,
              name: 'Frontend Developer',
              company: {
                brand: 'Faceb00k',
              },
              skill_tags: [],
            },
          },
          2: {
            items: {
              id: 2,
              name: 'Frontend Engineer',
              company: {
                brand: '4irBnb',
              },
              skill_tags: [{
                id: 1,
                name: 'Sketch',
              }, {
                id: 2,
                name: 'React.js',
              }],
            },
          },
        },
      }
    });

    subject = mount(
      <Provider store={store}>
        <JobContainer id="1" />
      </Provider>
    );

    Component = subject.find(JobContainer);
    JobComponent = Component.find(Job);
  });

  it('should be same as snapshot', () => {
    expect(JobComponent.html()).toMatchSnapshot();
  });
});
