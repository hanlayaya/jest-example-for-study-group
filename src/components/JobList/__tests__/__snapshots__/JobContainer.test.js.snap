// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container <Job /> should be same as snapshot 1`] = `
<Job
  dispatch={[Function]}
  id="1"
  job={
    Object {
      "company": Object {
        "brand": "Faceb00k",
      },
      "id": 1,
      "name": "Frontend Developer",
      "skill_tags": Array [],
    }
  }
>
  <div
    className="job"
  >
    <h1>
      Frontend Developer
    </h1>
    <h2>
      Faceb00k
    </h2>
    <div
      className="skill-tags"
    />
  </div>
</Job>
`;
